File: self-made/src/main/java/ooftf/com/widget/self/calendar/SimpleDateDrawModule.java
package ooftf.com.widget.self.calendar;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;

import java.util.Calendar;

import tf.oof.com.service.utils.CanvasUtil;
import tf.oof.com.service.utils.DateUtil;

/**
 * Created by master on 2017/8/10 0010.
 */

public class SimpleDateDrawModule implements DateDrawModule {
    Paint mDatePaint;
    Paint mPaintTodayBG;
    SimpleDateDrawModule(){
        mDatePaint = new Paint();
        mDatePaint.setAntiAlias(true);
        mPaintTodayBG = new Paint();
        mPaintTodayBG.setAntiAlias(true);
        mPaintTodayBG.setColor(Color.parseColor("#2EA7E0"));
    }

    @Override
    public void draw(Canvas canvas, Calendar current, float cx, float cy, int compareMonth,float width,float height) {
        mDatePaint.setTextSize(Math.min(width, height) / 3);
        String day = String.valueOf(getDay(current));
        if (compareMonth == 0) {
            if (DateUtil.compare(current, Calendar.getInstance().getInstance(), Calendar.DAY_OF_MONTH) == 0) {
                //当前日期
                mDatePaint.setColor(Color.parseColor("#ff0000"));
                canvas.drawCircle(cx, cy, Math.min(width, height) / 3, mPaintTodayBG);
            } else {
                //非当前日期
                mDatePaint.setColor(Color.parseColor("#000000"));
            }
        } else {
            mDatePaint.setColor(Color.parseColor("#999999"));
        }

        CanvasUtil.drawText(day,cx,cy,canvas,mDatePaint);
    }
    public int getDay(Calendar calendar) {
        return calendar.get(Calendar.DAY_OF_MONTH);
    }
}

File: self-made/src/main/java/ooftf/com/widget/self/calendar/SimpleDateDrawModule.kt
package ooftf.com.widget.self.calendar

import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint

import java.util.Calendar

import tf.oof.com.service.utils.CanvasUtil
import tf.oof.com.service.utils.DateUtil

/**
 * Created by master on 2017/8/10 0010.
 */

class SimpleDateDrawModule internal constructor() : DateDrawModule {
    internal var mDatePaint: Paint
    internal var mPaintTodayBG: Paint

    init {
        mDatePaint = Paint()
        mDatePaint.isAntiAlias = true
        mPaintTodayBG = Paint()
        mPaintTodayBG.isAntiAlias = true
        mPaintTodayBG.color = Color.parseColor("#2EA7E0")
    }

    override fun draw(canvas: Canvas, current: Calendar, cx: Float, cy: Float, compareMonth: Int, width: Float, height: Float) {
        val day = getDay(current).toString()
        if (compareMonth == 0) {
            if (DateUtil.compare(current, Calendar.getInstance(), Calendar.DAY_OF_MONTH) == 0) {
                //当前日期
                mDatePaint.color = Color.parseColor("#ff0000")
                canvas.drawCircle(cx, cy, Math.min(width, height) / 3, mPaintTodayBG)
            } else {
                //非当前日期
                mDatePaint.color = Color.parseColor("#000000")
            }
        } else {
            mDatePaint.color = Color.parseColor("#999999")
        }

        CanvasUtil.drawText(day, cx, cy, canvas, mDatePaint)
    }

    fun getDay(calendar: Calendar): Int {
        return calendar.get(Calendar.DAY_OF_MONTH)
    }

    override fun drawHeader(canvas: Canvas, value: String, index:Int,cx: Float, cy: Float, width: Float, height: Float) {
        mDatePaint.textSize = Math.min(width, height) / 3
        CanvasUtil.drawText(value, cx , cy, canvas, mDatePaint)
    }
}
